ddtrace/tracer/metrics_test.go

```diff
@@ -87,6 +89,125 @@ func TestEnqueuedTracesHealthMetric(t *testing.T) {
+			if slices.Equal(c.Tags(), []string{"integration:manual"}) {
+				return
+			}
+		}
+		assert.Fail("expected integration:manual tag in spans_started")
+	})
+
+	t.Run("other_source", func(t *testing.T) {
+		tg.Reset()
+		assert := assert.New(t)
+		tracer, _, _, stop := startTestTracer(t, withStatsdClient(&tg))
+		defer stop()
+
+		sp := tracer.StartSpan("operation", Tag(ext.Component, "contrib"))
+		defer sp.Finish()
+
+		tg.Wait(assert, 1, 100*time.Millisecond)
+
+		counts := tg.Counts()
+		assert.Equal(counts["datadog.tracer.spans_started"], int64(1))
+		for _, c := range statsdtest.FilterCallsByName(tg.CountCalls(), "datadog.tracer.spans_started") {
+			if slices.Equal(c.Tags(), []string{"integration:contrib"}) {
+				return
+			}
+		}
+		assert.Fail("expected integration:contrib tag in spans_started")
+
+	})
+}
+
+func TestSpansFinishedTags(t *testing.T) {
+	var tg statsdtest.TestStatsdClient
+
+	defer func(old time.Duration) { statsInterval = old }(statsInterval)
+	statsInterval = time.Millisecond
+
+	t.Run("default", func(t *testing.T) {
+		assert := assert.New(t)
+		tracer, _, _, stop := startTestTracer(t, withStatsdClient(&tg))
+		defer stop()
+
+		tracer.StartSpan("operation").Finish()
+		tg.Wait(assert, 1, 100*time.Millisecond)
+
+		counts := tg.Counts()
+		assert.Equal(counts["datadog.tracer.spans_finished"], int64(1))
+		for _, c := range statsdtest.FilterCallsByName(tg.CountCalls(), "datadog.tracer.spans_finished") {
+			if slices.Equal(c.Tags(), []string{"integration:manual"}) {
+				return
+			}
+		}
+		assert.Fail("expected integration:manual tag in spans_finished")
+	})
+
+	t.Run("other_source", func(t *testing.T) {
+		tg.Reset()
+		assert := assert.New(t)
+		tracer, _, _, stop := startTestTracer(t, withStatsdClient(&tg))
+		defer stop()
+
+		tracer.StartSpan("operation", Tag(ext.Component, "contrib")).Finish()
+
+		tg.Wait(assert, 1, 100*time.Millisecond)
+
+		counts := tg.Counts()
+		assert.Equal(counts["datadog.tracer.spans_finished"], int64(1))
+		for _, c := range statsdtest.FilterCallsByName(tg.CountCalls(), "datadog.tracer.spans_finished") {
+			if slices.Equal(c.Tags(), []string{"integration:contrib"}) {
+				return
+			}
+		}
+		assert.Fail("expected integration:contrib tag in spans_finished")
+
+	})
+}
+
+func TestHealthMetricRaces(t *testing.T) {
+	assert := assert.New(t)
+
+	defer func(old time.Duration) { statsInterval = old }(statsInterval)
+	statsInterval = time.Millisecond
+
+	var tg statsdtest.TestStatsdClient
+	tracer, _, flush, stop := startTestTracer(t, withStatsdClient(&tg))
+	defer stop()
+
+	for range 5 {
+		go func() {
+			sp := tracer.StartSpan("operation")
+			time.Sleep(100 * time.Millisecond)
+			sp.Finish()
+		}()
+	}
+	flush(5)
+	tg.Wait(assert, 2, 100*time.Millisecond)
+
+	counts := tg.Counts()
+	assert.Equal(int64(5), counts["datadog.tracer.spans_started"])
+	assert.Equal(int64(5), counts["datadog.tracer.spans_finished"])
+
```

Comment:
Can we also check the values on t.spansStarted.spans and t.spansFinished.spans? (They should be at 0 now, right?)